src/app/components/service/service.component.ts

```diff
@@ -221,4 +246,65 @@ export class ServiceComponent implements OnInit, OnDestroy {
         return 'g_translated'
     }
   }
+
+  updateMessage(data: EmittedData, id: string): void {
+    const { event, index } = data
+    const value = (event.target as HTMLTextAreaElement).innerText
+
+    const translation = this.translations$()[index].clone()
+    translation.messages = this.translations$()[index].messages.filter((message) => {
+      if (message.id === id) {
+        this.changes = message.message !== value ? true : false
```

Comment:
What type does "message.message !== value" return? :)

```suggestion
        this.changes = message.message !== value
```